mono/sgen/sgen-gc.c

```diff
@@ -1777,13 +1778,16 @@ collect_nursery (const char *reason, gboolean is_overflow)
 	sgen_client_pinning_end ();
 
 	remset.start_scan_remsets ();
+	TV_GETTIME (btv);
+	start_scan_remsets = TV_ELAPSED (atv, btv);
```

Comment:
You mentioned this could take unreasonable amounts of time. Maybe we should log it explicitly and not just cumulated with scan_remsets time.